arangod/Pregel/PregelFeature.cpp

```diff
@@ -162,10 +197,15 @@ std::pair<Result, uint64_t> PregelFeature::startExecution(
 
         if (!coll->isSmart()) {
           std::vector<std::string> eKeys = coll->shardKeys();
-          if (eKeys.size() != 1 || eKeys[0] != "vertex") {
+          if (eKeys.size() != 1 ||  params.get(shardKeyAttributeKey).compareString(eKeys[0])) {
```

Comment:
I think changes would only be required here:
1) get shardKeyAttribute value from job data if specified. If not specified, use "vertex"
2) compare actual shard key with shardKeyAttribute devised in 1)